lab_11/main.c

```diff
@@ -0,0 +1,174 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include "lib.h"
+
+int main()
+{
+    vehicle_t vehicle;
+    bool isVehicleCreated = false;
+    char choice = '\0';
+
+    memset(&vehicle, 0, sizeof(vehicle));
+
+    while (choice != 'q')
+    {
+        printf("Choose an option:\n");
+
+        if (!isVehicleCreated)
+        {
+            printf("1. Add Vehicle\n");
+        }
+        if (isVehicleCreated)
+        {
+            printf("2. Modify Vehicle\n");
+            printf("3. Modify Owner\n");
+            printf("4. Add Service\n");
+            printf("5. Clear Data\n");
+            printf("6. Display\n");
+        }
+        printf("q. Quit\n");
+        printf("Enter your choice: ");
+        if (scanf(" %c", &choice) != 1)
+        {
+            printf("Invalid input.\n");
+        }
+        getchar();
+
+        switch (choice)
+        {
+        case '1':
+            if (isVehicleCreated)
+            {
+                printf("Vehicle already exists.\n");
+                break;
+            }
+            {
+                owner_t owner;
+
+                printf("Enter the owner's name: ");
+                scanf(" %99[^\n]", owner.name);
+
+                printf("Enter the owner's ID: ");
+                scanf(" %99[^\n]", owner.id);
```

Comment:
There is some inconsistency, in the header file you have defined some constants, but here you use ‘magic numbers’.
In the following labs, we'll look at using fgets and string parsing.

Reply:
I had problem with input of string with spaces, and found this as temp solution, I will think about refactoring this